### PR TITLE
Link messages to respective conversation

### PR DESCRIPTION
**Pull request by elayabharath**
+ Added a default loading state, placeholder text when no messages are present
+ Added a link icon to the text message

Closes https://github.com/larksystems/Katikati-Core/issues/452

### CODE REVIEW

**Review by marianamarasoiu — APPROVED** [review]
Thanks @elayabharath - this is really cool!

Amusingly, this isn't quite addressing the issue you link to! The issue is about showing the link on the sample messages tooltip, and taking the user to the conversation view filtered to that tag (so appending &include-tags="tag-ID to the conversations URL). Your work here is going further by making every message a link, which is even cooler, but if possible I'd like to also have the filtered view link.

So, can you follow this PR up with another one that:
- makes "sample messages for tag" also a link button, that takes the user to the conversations view filtered by that tag
- the link for each message also includes the filtered view for that tag (so the user can easily find other similar messages if they wish)

I've edited the issue to include the above. thanks a lot, and LGTM on this part!

File: webapp/lib/app/configurator/tags/view.dart [hunk 1]
```diff
@@ -6,6 +6,7 @@ import 'package:dnd/dnd.dart' as dnd;
 import 'package:katikati_ui_lib/components/accordion/accordion.dart';
 import 'package:katikati_ui_lib/components/editable/editable_text.dart';
 import 'package:katikati_ui_lib/components/button/button.dart';
+import 'package:katikati_ui_lib/components/model/model.g.dart';
```
Comment (marianamarasoiu) [inline webapp/lib/app/configurator/tags/view.dart:9]: Nit: don't import the generated model directly, but instead import the wrapper `model.dart`
```suggestion
import 'package:katikati_ui_lib/components/model/model.dart';
```

**Comment by elayabharath** [comment]
Ahhh this makes sense - sorry for not understanding it in full. But added them both. Could you take a look at it one more time? Thanks!

File: webapp/lib/app/configurator/tags/view.dart [hunk 2]
```diff
@@ -257,9 +260,16 @@ class SampleMessagesTooltip {
     }
   }
 
-  String _conversationLinkFromMessageID(String messageID) {
-    var conversationID = messageID.replaceAll('nook-message-', '').substring(0, 52);
-    return "/converse/index.html?conversation-id=${conversationID}";
+  String _conversationLinkFromMessageID({String messageId, String tagId}) {
+    Map<String, String> queryParams = {};
+    if(messageId != null) {
```
Comment (marianamarasoiu) [inline webapp/lib/app/configurator/tags/view.dart:265]: nit:
```suggestion
    if (messageId != null) {
```

File: webapp/lib/app/configurator/tags/view.dart [hunk 3]
```diff
@@ -257,9 +260,16 @@ class SampleMessagesTooltip {
...
+      queryParams["conversation-id"] = messageId.replaceAll('nook-message-', '').substring(0, 52);
+    }
+    if(tagId != null) {
```
Comment (marianamarasoiu) [inline webapp/lib/app/configurator/tags/view.dart:268]:
```suggestion
    if (tagId != null) {
```

File: webapp/lib/app/configurator/tags/view.dart [hunk 4]
```diff
@@ -257,9 +260,16 @@ class SampleMessagesTooltip {
     }
   }
 
-  String _conversationLinkFromMessageID(String messageID) {
-    var conversationID = messageID.replaceAll('nook-message-', '').substring(0, 52);
-    return "/converse/index.html?conversation-id=${conversationID}";
+  String _conversationLinkFromMessageID({String messageId, String tagId}) {
```
Comment (marianamarasoiu) [inline webapp/lib/app/configurator/tags/view.dart:263]: I think we need to rename this now, maybe something like: `_linkToFilteredConversationView`?

**Review by marianamarasoiu — APPROVED** [review]
you're quick! thanks @elayabharath  - LGTM with a couple of small comments

**Pull request merged**